src/mapping/RadialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,207 @@
+    MultiscaleType type,
+    MultiscaleAxis axis)
+    : Mapping(constraint, dimensions, false, Mapping::InitialGuessRequirement::None),
+      _type(type), _axis(axis)
+{
+  setInputRequirement(Mapping::MeshRequirement::VERTEX);
+  setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+}
+
+void RadialGeoMultiscaleMapping::computeMapping()
+{
+  PRECICE_TRACE(output()->vertices().size());
+
+  PRECICE_ASSERT(input().get() != nullptr);
+  PRECICE_ASSERT(output().get() != nullptr);
+
+  size_t const inSize  = input()->vertices().size();
+  size_t const outSize = output()->vertices().size();
+
+  int effectiveCoordinate = _axis;
+  PRECICE_ASSERT(effectiveCoordinate == 0 ||
+                     effectiveCoordinate == 1 ||
+                     effectiveCoordinate == 2,
+                 "Unknown multiscale axis type.")
+
+  // determine principle axis midpoints as borders to assign 3D vertices to their respective 1D vertex
+  Eigen::VectorXd axisMidpoints(inSize);
+  _axisMidpoints        = axisMidpoints;
+  auto ordered_vertices = input()->vertices();
+  std::sort(ordered_vertices.begin(), ordered_vertices.end(),
+            [effectiveCoordinate](const mesh::Vertex &a, const mesh::Vertex &b) {
+              return a.getCoords()[effectiveCoordinate] < b.getCoords()[effectiveCoordinate];
+            });
+  for (size_t i = 0; i < (inSize - 1); i++) {
+    auto axisPositionCurrent = ordered_vertices[i].getCoords()[effectiveCoordinate];
+    auto axisPositionNext    = ordered_vertices[i + 1].getCoords()[effectiveCoordinate];
+    _axisMidpoints(i)        = (axisPositionCurrent + axisPositionNext) / 2;
+  }
+  _axisMidpoints(inSize - 1) = std::numeric_limits<double>::max(); // large number, such that vertices after the last midpoint are still assigned
+
+  if (getConstraint() == CONSISTENT) {
+    PRECICE_DEBUG("Compute consistent mapping");
+    if (_type == SPREAD) {
+      /*
+        3D vertices are projected onto the 1D axis and the data is then mapped
+        to the nearest neighbors of the 1D vertices in projection space.
+      */
+      for (size_t i = 0; i < outSize; i++) {
+        auto vertexCoord = output()->vertices()[i].getCoords()[effectiveCoordinate];
+        int  index       = 0;
+        while (vertexCoord > _axisMidpoints(index)) {
+          index++;
+          PRECICE_ASSERT(index < static_cast<int>(inSize));
+        }
+        _vertexIndicesSpread.push_back(index);
+      }
+    } else {
+      PRECICE_ASSERT(_type == COLLECT)
+      /*
+        3D vertices are projected onto the 1D axis and the data is then mapped
+        to (and averaged at) the nearest 1D vertex in projection space.
+      */
+      size_t const inSize  = input()->vertices().size();
+      size_t const outSize = output()->vertices().size();
+
+      // determine principle axis midpoints as borders to assign 3D vertices to their respective 1D vertex
+      Eigen::VectorXd axisMidpoints(outSize);
+      _axisMidpoints = axisMidpoints;
```

Comment:
Do we need the `axisMidpoints`? Looks like we can just resize the `_axisMidpoints`.

Reply:
I think we might not even need the `_axisMidpoints`, i.e., we don't use them in any other place, so I think we can stick with them not being class members.